src/js/modules/FormFeedback.js

```diff
@@ -0,0 +1,231 @@
+import React, { Component } from 'react';
+import { defineMessages, injectIntl } from 'react-intl';
+import { postFeedback } from 'js/helpers/fetchData';
+import { logEvent } from 'js/helpers/googleAnalytics';
+import SectionHeaderSerif from 'js/modules/SectionHeaderSerif';
+
+import emojiDisappointed from 'images/emojis/disappointed.png';
+import emojiSad from 'images/emojis/sad.png';
+import emojiNeutral from 'images/emojis/neutral.png';
+import emojiSlightlySmiling from 'images/emojis/slightly_smiling.png';
+import emojiGrinning from 'images/emojis/grinning.png';
+
+const i18nMessages = defineMessages({
+  step1title: {
+    id: 'FormFeedback.step1.title',
+    defaultMessage: 'How satisfied are you with this page?',
+  },
+  step2titlea: {
+    id: 'FormFeedback.step2.title.a',
+    defaultMessage: 'Thank you!',
+  },
+  step2titleb: {
+    id: 'FormFeedback.step2.title.b',
+    defaultMessage: 'How can we make this page better?',
+  },
+  step3title: {
+    id: 'FormFeedback.step3.title',
+    defaultMessage: 'Thank you for sharing feedback!',
+  },
+  step3button: {
+    id: 'FormFeedback.step3.button',
+    defaultMessage: 'Give more feedback',
+  },
+  emojiDisappointed: {
+    id: 'FormFeedback.emojiDisappointed.altText',
+    defaultMessage: 'Disappointed',
+  },
+  emojiSad: {
+    id: 'FormFeedback.emojiSad.altText',
+    defaultMessage: 'Sad',
+  },
+  emojiNeutral: {
+    id: 'FormFeedback.emojiNeutral.altText',
+    defaultMessage: 'Neutral',
+  },
+  emojiSlightlySmiling: {
+    id: 'FormFeedback.emojiSlightlySmiling.altText',
+    defaultMessage: 'Slightly Smiling',
+  },
+  emojiGrinning: {
+    id: 'FormFeedback.emojiGrinning.altText',
+    defaultMessage: 'Grinning',
+  },
+});
+
+
+class FormFeedback extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      step: 1,
+      loading: false,
+      emoji: null,
+      feedback: null,
+      error: null
+    };
+  }
+
+  logEvent(action, label) {
```

Comment:
I'm confused by this. It seems like you are defining `logEvent` as a function that calls the `logEvent` function you imported from helpers. Maybe just using different naming would help? Also, why not use the arrow function syntax like you did below?

Reply:
Yea, renaming might help -- basically I'm using the component scoped logEvent to call the more generic helper logEvent with some automatic behavior baked in -- passing the step and form and form name. I can make this better for sure.

I didn't make it an arrow function because it's not an event handler so I don't need to create a this scope for it.

Reply:
Implemented a generic logFormEvent in our googleAnalytic.js helper functions. the class method logEvent now calls the helper method logFormEvent. Hopefully this is clearer .